psalm.xml

```diff
@@ -26,5 +26,10 @@
         <ImplementedReturnTypeMismatch errorLevel="suppress"/>
         <ImplicitToStringCast errorLevel="suppress"/>
         <UndefinedMagicMethod errorLevel="suppress"/>
+        <UndefinedMethod>
+            <errorLevel type="suppress">
+                <referencedMethod name="Symfony\Component\Config\Definition\Builder\TreeBuilder::root"/>
+            </errorLevel>
+        </UndefinedMethod>
```

Comment:
Would you mind explaining why is this necessary?

Reply:
because psalm is complaining that a method is called that does not exist: https://github.com/propelorm/Propel2/runs/3067726361
(but it exists in older symfony versions)

Reply:
But it did not complain before, when it was called in the `if`?
On a semi-related note: where did you see the deprecation notice?

Reply:
Instead of changing the psalm config, I could add this to the else branch:

```php
assert(method_exists($treeBuilder, 'root'));
```

Reply:
Interesting. But I don't think that's necessary. You are right, swapping the rule from stan to psalm seems alright.